portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -60,53 +62,49 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
             comments.add(new Comment(name, content));
         }
 
-        Gson gson = new Gson();
-        String json = gson.toJson(comments);
+        // Gson gson = new Gson();
+        // String json = gson.toJson(comments);
         response.setContentType("application/json");
-        response.getWriter().println(json);
+        response.getWriter().println(convertToJsonUsingGson(comments));
     }
 
     /**
      * Obtain parameters of the HTML form and creates an entity in Datastore.
      */
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        String name = null;
-        String content = null;
-        boolean owo = false;
-        boolean uwu = false;
-
-        try {
-            name = getParameter(request, "name-input", "");
-            content = getParameter(request, "text-input", "");
-            owo = getBooleanParameter(request, "owo", false);
-            uwu = getBooleanParameter(request, "uwu", false);
-            DataServlet.numberOfShowedComments = getIntParameter(request, "num-input", 3);
-        } catch (NumberFormatException nfe) {
-            response.sendError(400, "Comment amount must be a number");
-            return;
-        } catch (Exception e){
-            response.sendError(500, "Error");
-            return;
+        ArrayList<Integer> errors = new ArrayList<>();
+        String name = getParameter(request, "name-input", "");
+        String content = getParameter(request, "text-input", "");
+        boolean owo = getBooleanParameter(request, "owo", false);
+        boolean uwu = getBooleanParameter(request, "uwu", false);
+        DataServlet.numberOfShowedComments = getIntParameter(request, errors, "num-input", 3);
+        
+        String commentConcatenated = owoUwuTransformer(content, owo, uwu);
+
+        if(errors.size() == 0){
+            Entity commentEntity = new Entity("Comment");
+            commentEntity.setProperty("name", name);
+            commentEntity.setProperty("content", commentConcatenated);
```

Comment:
Similar to the MemePost, the comment class (I believe you have a class for this?) should have a getDatabaseEntity method on it and this should be just `new Comment(name, commentConcatenated).getDatabaseEntity()`

Reply:
Got it.